### PR TITLE
BugFix: fix Apple Catalyst flags

### LINKED ISSUE #8387: [bug] Catalyst subsystem setting should be all lowercase
https://github.com/conan-io/conan/pull/8264/files

The new Macos subsystem `Catalyst` in settings should be all lowercase to be consistent with existing subsystems. 

https://github.com/SSE4/conan/blob/ff7de2e5b2551b0cc69970e8bb6edf43b6760ba0/conans/client/conf/__init__.py#L41
` subsystem: [None, "Catalyst"]`

https://github.com/SSE4/conan/blob/ff7de2e5b2551b0cc69970e8bb6edf43b6760ba0/conans/client/conf/__init__.py#L31
` subsystem: [None, cygwin, msys, msys2, wsl]`

### LINKED ISSUE #8388: [question] How to build for Mac Catalyst?
1.33 with catalyst support is out and I tried it right away. Maybe with a bit naive first try. :-)

### Environment Details (include every applicable attribute)
  * Operating System+version: Catalina
  * Compiler+version: Apple clang version 12.0.0 (clang-1200.0.32.28)
  * Conan version: 1.33
  * Python version:  3.7.4

### Steps to reproduce (Include if Applicable)
conan install xmlsec/1.2.31@ --build --profile ./conan/profiles/mac_catalyst

### Error
from config.log

>configure:4198: $? = 0
configure:4187: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -V >&5
clang: error: argument to '-V' is missing (expected 1 value)
clang: error: no input files
configure:4198: $? = 1
configure:4187: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -qversion >&5
**clang: error: unknown argument '-qversion'; did you mean '--version'?**
clang: error: no input files
configure:4198: $? = 1
configure:4218: checking whether the C compiler works
configure:4240: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -target=x86_64-apple-ios-macabi -O3 --sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fPIC -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -DNDEBUG -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -target=x86_64-apple-ios-macabi --sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 conftest.c  >&5
**clang: error: unknown argument '-target=x86_64-apple-ios-macabi'; did you mean '--target=x86_64-apple-ios-macabi'?
clang: error: unknown argument '-target=x86_64-apple-ios-macabi'; did you mean '--target=x86_64-apple-ios-macabi'?**
configure:4244: $? = 1

______
???
configure:4240: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang 
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-arch x86_64
-target=x86_64-apple-ios-macabi
-O3
--sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-fPIC 
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-arch x86_64 -DNDEBUG
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk 
-arch x86_64
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-arch x86_64 -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-arch x86_64
-target=x86_64-apple-ios-macabi
--sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk 
-arch x86_64 conftest.c  >&5

arch=x86_64
build_type=Release
compiler=apple-clang
compiler.cppstd=14
compiler.libcxx=libc++
compiler.version=12.0
os=Macos
os.subsystem=Catalyst
os_build=Macos
[options]
[build_requires]
*: darwin-toolchain/1.0.8@theodelrieu/stable
[env]

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
conan install xmlsec/1.2.31@ --build --profile ./conan/profiles/mac_catalyst
Configuration:
[settings]
arch=x86_64
build_type=Release
compiler=apple-clang
compiler.cppstd=14
compiler.libcxx=libc++
compiler.version=12.0
os=Macos
os.subsystem=Catalyst
os_build=Macos
[options]
[build_requires]
*: darwin-toolchain/1.0.8@theodelrieu/stable
[env]

libiconv/1.16: Forced build from source
openssl/1.1.1h: Forced build from source
zlib/1.2.11: Forced build from source
libxml2/2.9.10: Forced build from source
xmlsec/1.2.31: Forced build from source
darwin-toolchain/1.0.8@theodelrieu/stable: Forced build from source
m4/1.4.18: Forced build from source
pkgconf/1.7.3: Forced build from source
autoconf/2.69: Forced build from source
automake/1.16.3: Forced build from source
libtool/2.4.6: Forced build from source
ninja/1.10.2: Forced build from source
meson/0.55.1: Forced build from source
Installing package: xmlsec/1.2.31
Requirements
    libiconv/1.16 from 'conan-center' - Cache
    libxml2/2.9.10 from 'conan-center' - Cache
    openssl/1.1.1h from 'conan-center' - Cache
    xmlsec/1.2.31 from 'conan-center' - Cache
    zlib/1.2.11 from 'conan-center' - Cache
Packages
    libiconv/1.16:9d3239a97b703c82f38d9e91676ca971d7eb7701 - Build
    libxml2/2.9.10:a2fcfc5995ddeefb1ff988dbc9aa90b909d43ed5 - Build
    openssl/1.1.1h:9d3239a97b703c82f38d9e91676ca971d7eb7701 - Build
    xmlsec/1.2.31:6c9b5886a0769be2cb726e1b2e87b7bc5c550492 - Build
    zlib/1.2.11:9d3239a97b703c82f38d9e91676ca971d7eb7701 - Build
Build requirements
    autoconf/2.69 from 'conan-center' - Cache
    automake/1.16.3 from 'conan-center' - Cache
    darwin-toolchain/1.0.8@theodelrieu/stable from 'conan-center' - Cache
    libtool/2.4.6 from 'conan-center' - Cache
    m4/1.4.18 from 'conan-center' - Cache
    meson/0.55.1 from 'conan-center' - Cache
    ninja/1.10.2 from 'conan-center' - Cache
    pkgconf/1.7.3 from 'conan-center' - Cache
Build requirements packages
    autoconf/2.69:923c5310a94256af03d94c969eebbbadac513c42 - Build
    automake/1.16.3:8198ae767c7bf455a38775936906b3e982c035ec - Build
    darwin-toolchain/1.0.8@theodelrieu/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Build
    libtool/2.4.6:072e844e71af4f1016fea59345350a58177d9979 - Build
    m4/1.4.18:3bc8a0eb26ab34ed4958480d81dfd67c24f74ed4 - Build
    meson/0.55.1:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Build
    ninja/1.10.2:3f3351f8c9c8848e68a09e5f612a66c58eb79473 - Build
    pkgconf/1.7.3:9d3239a97b703c82f38d9e91676ca971d7eb7701 - Build

Installing (downloading, building) binaries...
darwin-toolchain/1.0.8@theodelrieu/stable: Copying sources to build folder
darwin-toolchain/1.0.8@theodelrieu/stable: Building your package in /Users/dmn/.conan/data/darwin-toolchain/1.0.8/theodelrieu/stable/build/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
darwin-toolchain/1.0.8@theodelrieu/stable: Generator txt created conanbuildinfo.txt
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] 'fPIC' option not found
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK
darwin-toolchain/1.0.8@theodelrieu/stable: Calling build()
darwin-toolchain/1.0.8@theodelrieu/stable: WARN: This conanfile has no build step
darwin-toolchain/1.0.8@theodelrieu/stable: Package '5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9' built
darwin-toolchain/1.0.8@theodelrieu/stable: Build folder /Users/dmn/.conan/data/darwin-toolchain/1.0.8/theodelrieu/stable/build/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
darwin-toolchain/1.0.8@theodelrieu/stable: Generated conaninfo.txt
darwin-toolchain/1.0.8@theodelrieu/stable: Generated conanbuildinfo.txt
darwin-toolchain/1.0.8@theodelrieu/stable: Generating the package
darwin-toolchain/1.0.8@theodelrieu/stable: Package folder /Users/dmn/.conan/data/darwin-toolchain/1.0.8/theodelrieu/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
darwin-toolchain/1.0.8@theodelrieu/stable: Calling package()
[HOOK - conan-center.py] post_package(): ERROR: [PACKAGE LICENSE (KB-H012)] No 'licenses' folder found in package: /Users/dmn/.conan/data/darwin-toolchain/1.0.8/theodelrieu/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9  (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H012)
[HOOK - conan-center.py] post_package(): ERROR: [DEFAULT PACKAGE LAYOUT (KB-H013)] Unknown file 'darwin-toolchain.cmake' in the package (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H013)
[HOOK - conan-center.py] post_package(): [DEFAULT PACKAGE LAYOUT (KB-H013)] If you are trying to package a tool put all the contents under the 'bin' folder
[HOOK - conan-center.py] post_package(): ERROR: [MATCHING CONFIGURATION (KB-H014)] Empty package (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H014)
[HOOK - conan-center.py] post_package(): ERROR: [MATCHING CONFIGURATION (KB-H014)] Packaged artifacts does not match the settings used: os=Macos, compiler=apple-clang (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H014)
[HOOK - conan-center.py] post_package(): [SHARED ARTIFACTS (KB-H015)] OK
[HOOK - conan-center.py] post_package(): [PC-FILES (KB-H020)] OK
[HOOK - conan-center.py] post_package(): [CMAKE-MODULES-CONFIG-FILES (KB-H016)] OK
[HOOK - conan-center.py] post_package(): [PDB FILES NOT ALLOWED (KB-H017)] OK
[HOOK - conan-center.py] post_package(): [LIBTOOL FILES PRESENCE (KB-H018)] OK
[HOOK - conan-center.py] post_package(): [MS RUNTIME FILES (KB-H021)] OK
darwin-toolchain/1.0.8@theodelrieu/stable package(): Packaged 1 '.cmake' file: darwin-toolchain.cmake
darwin-toolchain/1.0.8@theodelrieu/stable: Package '5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9' created
darwin-toolchain/1.0.8@theodelrieu/stable: Created package revision c093366e06ad120604635abd2f99a5d2
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
libiconv/1.16: Applying build-requirement: darwin-toolchain/1.0.8@theodelrieu/stable
libiconv/1.16: WARN: Build folder is dirty, removing it: /Users/dmn/.conan/data/libiconv/1.16/_/_/build/9d3239a97b703c82f38d9e91676ca971d7eb7701
libiconv/1.16: Copying sources to build folder
libiconv/1.16: Building your package in /Users/dmn/.conan/data/libiconv/1.16/_/_/build/9d3239a97b703c82f38d9e91676ca971d7eb7701
libiconv/1.16: Generator txt created conanbuildinfo.txt
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK. 'fPIC' option found and apparently well managed
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK
libiconv/1.16: Calling build()
libiconv/1.16: Calling:
 > ./configure '--enable-static' '--disable-shared' '--prefix=/Users/dmn/.conan/data/libiconv/1.16/_/_/package/9d3239a97b703c82f38d9e91676ca971d7eb7701' '--bindir=${prefix}/bin' '--sbindir=${prefix}/bin' '--libexecdir=${prefix}/bin' '--libdir=${prefix}/lib' '--includedir=${prefix}/include' '--oldincludedir=${prefix}/include' '--datarootdir=${prefix}/share'
checking for a BSD-compatible install... /usr/local/bin/ginstall -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/local/bin/gmkdir -p
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make sets $(MAKE)... (cached) yes
checking for gcc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
checking whether the C compiler works... no
configure: error: in `/Users/dmn/.conan/data/libiconv/1.16/_/_/build/9d3239a97b703c82f38d9e91676ca971d7eb7701/source_subfolder':
configure: error: C compiler cannot create executables
See `config.log' for more details
libiconv/1.16:
libiconv/1.16: ERROR: Package '9d3239a97b703c82f38d9e91676ca971d7eb7701' build failed
libiconv/1.16: WARN: Build folder /Users/dmn/.conan/data/libiconv/1.16/_/_/build/9d3239a97b703c82f38d9e91676ca971d7eb7701
ERROR: libiconv/1.16: Error in build() method, line 127
	autotools = self._configure_autotools()
while calling '_configure_autotools', line 117
	self._autotools.configure(args=configure_args, host=host, build=build)
	ConanException: Error 77 while executing ./configure '--enable-static' '--disable-shared' '--prefix=/Users/dmn/.conan/data/libiconv/1.16/_/_/package/9d3239a97b703c82f38d9e91676ca971d7eb7701' '--bindir=${prefix}/bin' '--sbindir=${prefix}/bin' '--libexecdir=${prefix}/bin' '--libdir=${prefix}/lib' '--includedir=${prefix}/include' '--oldincludedir=${prefix}/include' '--datarootdir=${prefix}/share'
```

### PR DESCRIPTION
**Pull request by SSE4**
closes: #8388
closes: #8387

fix several bugs spotted in Catalyst implementation:
- `os.subsystem` is now lower-case `catalyst` and no longer quoted in settings.yml, to match other values
- added `13.0` to `os.version` for Catalyst
- use `--target=x86_64-apple-ios-macabi` instead of `-target=x86_64-apple-ios-macabi` (two dashes)
- ensure `-mios-version-min` is passed for `AutoToolsBuildEnvironment`

Changelog: BugFix: Fix Apple Catalyst flags.
Changelog: Fix: Make the new Macos subsystem Catalyst lowercase to be consistent with existing subsystems.
Docs: https://github.com/conan-io/docs/pull/2009

- [x] Refer to the issue that supports this Pull Request.
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

**Review by memsharded — CHANGES_REQUESTED** [review]
please target it to release/1.33 branch, lets fix this into the next patch 1.33.1

**Comment by SSE4** [comment]
~~@memsharded seems like there is no such a branch~~ never mind, was looking for `release/1.33.1`

File: conans/client/build/compiler_flags.py [hunk 1]
```diff
@@ -47,8 +47,8 @@ def architecture_flag(settings):
         return ""
 
     if str(compiler) in ['gcc', 'apple-clang', 'clang', 'sun-cc']:
-        if str(the_os) == 'Macos' and str(subsystem) == 'Catalyst' and str(arch) == 'x86_64':
-            return '-target=x86_64-apple-ios-macabi'
+        if str(the_os) == 'Macos' and str(subsystem) == 'catalyst' and str(arch) == 'x86_64':
+            return '--target=x86_64-apple-ios-macabi'
```
Comment (memsharded) [inline conans/client/build/compiler_flags.py:51]: I know this is not from this PR, but I think this --target might conflict with the --target that the AutotoolsBuildEnvironment will add. Please have a look at it.

**Review by memsharded — APPROVED** [review]
Looking good, but please check the --target issue.

File: conans/client/build/compiler_flags.py (hunk 1 above)
Comment (SSE4) [reply conans/client/build/compiler_flags.py:51]: I don't think so, this one is a compiler flag, while AutoToolsBuildEnvironment only passes flag to the configure script itself

Comment (memsharded) [reply conans/client/build/compiler_flags.py:51]: I am not sure, this is being added in AutoToolsBuildEnvironment as flags:
```python
 def _configure_flags(self):
        ret = list(self._deps_cpp_info.cflags)
        arch_flag = architecture_flag(self._conanfile.settings)
        if arch_flag:
            ret.append(arch_flag)
```
Then ``configure()`` itself is also adding:
```python
 if target is not False:  # Skipped by user
            if target or self.target:  # User specified value or automatic
                triplet_args.append("--target=%s" % (target or self.target))
```

**Review by jgsogo — APPROVED** [review]

File: conans/client/build/compiler_flags.py (hunk 1 above)
Comment (memsharded) [reply conans/client/build/compiler_flags.py:51]: I can see that this is difficult to reproduce, as getting the --target is not easy unless building a graph with build_requires that will add the --target triplet, but still looks possible. Maybe we want to move forward and leave this as an open issue for later.

Comment (jgsogo) [reply conans/client/build/compiler_flags.py:51]: I see the `--target` from @SSE4 changes affecting the `configure` command-line call, while the existing `--target` could be added to the _flags_ that are populating the environment (hard to follow having a look to the sources). It looks like both can collide...

Comment (memsharded) [reply conans/client/build/compiler_flags.py:51]: Yes exactly. The only thing is that at the moment this is very unlikely, it will only happen when the --target is involved (not many times, only special build context builds) and the new catalyst. 

We are already starting to decouple this helper into 3 of them, I think we could figure this out in the new ones (merging this now)

Comment (jgsogo) [reply conans/client/build/compiler_flags.py:51]: I'm totally ok to forget about this issue now... and focus on how this information will be propagated from build-requires, or how to mix it when coming from different sources.
